### DISCUSSION TITLE
Punchy 0.0.4 Release Planning

### DISCUSSION THREAD

**Discussion by zicklag** [fishfight/punchy#177] [Planning]
Having a release planning discussion seemed to work out pretty well last time, so I'm staring one for 0.0.4, too.

**Comment by zicklag** [comment]
@erlend-sh @odecay While some of the time spent so far has been slow because I'd been away or not feeling well, I started working on scripting POC, ~11 days ago and I wanted to le you guys know you can tell me if other stuff should take priority at any point.

I think scripting could be important and impact how we design things like items and attacks moving forward, but it's also a complicated subject, and something I've personally been interested in getting working for years, so if it ends up taking more time than it's worth at the moment, feel free to let me know I should work on something else!

I'm pretty sure I'm getting close to wrapping up the ability to execute scripts in #147, and then it might move on to working with @odecay  and @64kramsystem to find out what's a good candidate for logic we could move into a script so I can figure out what kind of API support we need to make the script do something useful to us in the game.

I just wanted to keep the current status nice and transparent so that we can keep making progress on the things that count!

**Reply by erlend-sh** [reply]
> if [scripting] ends up taking more time than it's worth at the moment, feel free to let me know I should work on something else!

Good scripting support is strategically important to us because of our modding ambitions, so definitely feel free to keep pursuing it 👍

**Comment by 64kramsystem** [comment]
Things I plan to work on, if nobody else will take them, in no particular order:

- ~~#99~~: taken over by @zicklag with #203
- ~~#155~~: I've completed the original task, but an additional task was added, so it's up for the grabs
- ~~#145~~
- ~~Update #171~~
- ~~#85~~

I will only assign myself to the task(s) I'm currently working on, so anybody can take any without any problem (especially, newcomers).

**Reply by 64kramsystem** [reply]
Uh, I did actually assign myself already to some :sweat_smile:

**Reply by odecay** [reply]
I was intending to do Melee Weapons because I think it will play into the rest of the attack system stuff I'm going to work on, but I wouldn't be opposed to you taking it if you really want it or if I end up not having time. Its also low on my list so if you get to it first go for it.

**Reply by 64kramsystem** [reply]
> I was intending to do Melee Weapons because I think it will play into the rest of the attack system stuff I'm going to work on, but I wouldn't be opposed to you taking it if you really want it or if I end up not having time. Its also low on my list so if you get to it first go for it.

Sure, removed it from my list.

**Comment by odecay** [comment]
Heres my list, basically in order which I will work on them.
- #157 
- #152 
- #148 
- #179 
- #180

**Reply by odecay** [reply]
I've been sick for the majority of past week, hopefully will be able to check back up with what ya'll have been up to soon.

**Reply by odecay** [reply]
Getting back into working on attack system stuff as of today/yesterday.

**Comment by 64kramsystem** [comment]
I'm almost done with my tasks for the milestone. @erlend-sh which is the planned release date?

I'll see if I can take some Boss tasks, while working in parallel with @edgarssilva.

**Reply by erlend-sh** [reply]
Standard is 2 weeks after last release.

**Comment by zicklag** [comment]
I moved a group of my tasks to the next release, as I got most of my time taken up with the new state machine, but that was a lot to get done!

The biggest thing I've got left that _needs_ to happen before the next release is to figure out why gamepad controls are all busted ( #213  ), then I can help with anything else we need to get done before the next release, if there's anything it would be particularly useful for me to work on. ( let me know! )

Otherwise, I'm still working on porting my scripting stuff to a PR for `bevy_mod_js_scripting` and that's going well, so can keep working on that.

**Comment by odecay** [comment]
Is bors being weird? It seems like its stalled out on every pr with 
`bors Expected — Waiting for status to be reported`

edit: Maybe hanging is the wrong way to describe it but it feels like its been slow, I wonder if anything changed or I'm imagining it, I didn't notice logs saying it seemed to be taking longer :shrug:

**Reply by zicklag** [reply]
Oh, that's because we set bors to be required to merge PRs, so GitHub will mark a PR as not having passed checks until we run `bors merge` and it actually merges it.

It keeps us from accidentally merging it, but unfortunately makes every PR get the little orange dot until it's merged. :/

**Reply by edgarssilva** [reply]
> Oh, that's because we set bors to be required to merge PRs, so GitHub will mark a PR as not having passed checks until we run `bors merge` and it actually merges it.

That's right, a bit annoying yeah.

**Reply by odecay** [reply]
Ahhhh I see. Here I was just looking at it thinking "when is this last check going to finish" lol.

**Comment by odecay** [comment]
Okay yall I feel like I dropped the thread on this release. So far its mostly been the big refactor and adapting things to that. I think it left us in a much better state going forward.

I was unable to get through any of the major things I intended to work on this cycle and have been dragging my feet on getting a release out due to that, but I've also noticed some small gamefeel things post refactor that should be addressed before another release.

gamefeel things I noticed so far, both of these should be relatively simple fixes.
- due to in progress attack changes basic attack is less interesting since it is stationary, but without a new animation and without chained follow up hits, this could be easily fixed by changing it back to the flop attack for the release.
- State transition changes left knockback as lower priority than attacks which is the opposite of how it worked before. 

I will fix these up quickly, wondering if there is anything else we should consider before getting a release out.

**Reply by zicklag** [reply]
That sounds good to me. :+1: 

I don't think there's anything else that needs to get done before a next release.

**Reply by odecay** [reply]
Other thing i noticed is that the boss walks off the bottom of the screen often, I cant remember what happened with the idea to change fighter position to be based on their feet/bottom of sprite, did that happen?
